syntax = "proto3";

message ListTestsRequest {
    uint32 fixture_id = 1;
}

message ListTestsResponse {
    message TestInfo {
        uint32 id = 1;
        string name = 2;
    }
    repeated TestInfo tests = 1;
}

message RunTestRequest {
    uint32 fixture_id = 1;
    uint32 test_id = 2;
}

message RunTestResponse {
    uint64 status = 1;
}

message Request {
    oneof requests {
        ListTestsRequest list_tests = 1;
        RunTestRequest run_test = 2;
    }
}

message Response {
    oneof responses {
        ListTestsResponse list_tests = 1;
        RunTestResponse run_test = 2;
    }
}
